src/lib.rs

```diff
@@ -145,15 +190,128 @@ mod test_utils {
             .unwrap();
 
         #[cfg(not(any(feature = "stronghold", feature = "stronghold-storage")))]
-        crate::signing::set_signer(signer_type(), TestSigner {}).await;
+        crate::signing::set_signer(signer_type(), TestSigner::default()).await;
+
         #[cfg(any(feature = "stronghold", feature = "stronghold-storage"))]
         manager.set_stronghold_password("password").await.unwrap();
 
-        manager.store_mnemonic(signer_type(), None).await.unwrap();
-
         manager
     }
 
+    struct StorageTestCase {
+        storage_password: Option<String>,
+        storage: ManagerStorage,
+    }
+
+    enum ManagerTestCase {
+        Signer(SignerType),
+        Storage(StorageTestCase),
+    }
+
+    #[derive(PartialEq)]
+    pub enum TestType {
+        Signing,
+        Storage,
+        SigningAndStorage,
+    }
+
+    pub async fn with_account_manager<R: futures::Future<Output = ()>, F: Fn(AccountManager, SignerType) -> R>(
+        test_type: TestType,
+        cb: F,
+    ) {
+        let mut test_cases: Vec<ManagerTestCase> = Vec::new();
+
+        if test_type == TestType::Signing || test_type == TestType::SigningAndStorage {
+            test_cases.push(ManagerTestCase::Signer(SignerType::Custom("".to_string())));
+            #[cfg(feature = "stronghold")]
+            {
+                test_cases.push(ManagerTestCase::Signer(SignerType::Stronghold));
+            }
+        }
+
+        if test_type == TestType::Storage || test_type == TestType::SigningAndStorage {
+            // ---- Stronghold storage ----
+            #[cfg(feature = "stronghold-storage")]
+            test_cases.push(ManagerTestCase::Storage(StorageTestCase {
+                storage_password: None,
+                storage: ManagerStorage::Stronghold,
+            }));
+
+            test_cases.push(ManagerTestCase::Storage(StorageTestCase {
```

Comment:
Is this specific to `stronghold-storage` as well?

Reply:
yeah you're right, thanks for finding that :P